packages/connections/src/stores/connections-store.ts

```diff
@@ -26,13 +28,35 @@ export function createNewConnectionInfo(): ConnectionInfo {
     },
   };
 }
-
 export interface ConnectionStore {
   loadAll: () => Promise<ConnectionInfo[]>;
   save: (connectionInfo: ConnectionInfo) => Promise<void>;
   delete: (connectionInfo: ConnectionInfo) => Promise<void>;
 }
 
+function setAppNameParamIfMissing(
+  connectionString: string,
+  appName: string
+): string {
+  let connectionStringUrl;
+
+  try {
+    connectionStringUrl = new ConnectionString(connectionString);
+  } catch (e) {
+    //
+  }
+
+  if (!connectionStringUrl) {
+    return connectionString;
+  }
+
+  if (!connectionStringUrl.searchParams.has('appName')) {
+    connectionStringUrl.searchParams.set('appName', appName);
```

Comment:
totally optional, but since we have started being fairly consistent about this elsewhere:

```suggestion
  const searchParams = connectionStringUrl.typedSearchParams<MongoClientOptions>();
  if (!searchParams.has('appName')) {
    searchParams.set('appName', appName);
```

(with `import type { MongoClientOptions } from 'mongodb';` above)